docker-compose.yaml

```diff
@@ -0,0 +1,15 @@
+version: "3.9"
+services:
+  3dmmforever-build:
+    image: 3dmmforever-build
+    container_name: 3dmmforever-build
+    build: .

```

Comment:
This area here seems to do all the magic right?  Isnt it mounting the current dir of master to C:\3d and then how does command in the yaml work with the Dockerfile's entry point?

Im somewhat still learning docker-compose, so I was curious how this works in what you setup.  I would expect that the entrypoint in the Dockerfile would somehow call this command to compile 3dmm.  But maybe you can explain more to help me learn docker-compose more.

Thanks @ckoryom, good work on this for me to experiment on builds locally without having to ruin my environment.  This way I can keep it all in a container

Do the builds just go into a local builds folder after compiling with docker up?

Reply:
The key is in

```
    volumes:
      - .:C:\3d
```
Which shares the relative local host directory (to where the docker-compose.yaml file is located) with the container, after running you'd have the compiled result in your host machine

Reply:
Hey @davidrennet thanks for your review/comments!

The `build: .` area actually just tells `docker-compose` to use the` Dockerfile` in the current context.

As @sandman21dan  mentioned your current directory (where 3DMMForever is located) is mounted as a volume on the container at `C:\3d`. Everything that happens inside the container will change your local files (the compilation).

I added the Entrypoint in the Dockerfile in case you want to run the container without docker-compose and attach to it, so in that case you land directly with the VsDevCmd.bat

To your last question: "Do the builds just go into a local builds folder after compiling with docker up?" Absolutely!

Reply:
The only downside of this image is that it's really big because of windows-servercore takes a bunch of space, but also the VS Buildtools take aroud half of the size.

In the future it seems Microsoft will allow the Buildtools to be installed in windows-nanoserver (which is a much smaller image)